### ISSUE TITLE
MuTect2 Should Error if Almost All Somatic Variants Supported by Few Reads

### ISSUE THREAD

**Issue by DarioS** [broadinstitute/gatk#6674]
After doing somatic variant calling, the AD shows that the alternate allele is supported by three to five reads out of typically about eighty reads for almost all of the variants and the patient has about 10 times less SNVs reported in their VCF than all of the other patients with the same disease. Could MuTect2 have a QC failure status output if almost all of the variants reported are close to the limit of detection, which appears to be about three reads for MuTect2, looking at the second AD values in the cancer sample?

**Comment by lbergelson** [comment]
@DarioS That sounds like an impure tumor sample.   Isn't that normal in a lot of cancer types?

**Comment by fleharty** [comment]
@DarioS Are you running contamination checks on this sample first?  Did that not fail your QC?

It's difficult for us to provide a failure QC metric like this because there are so many different cancer types, each with their own behaviors.  The fact that variants are all near their limit of detection is not necessarily a failure of M2 or even your sample necessarily.

Instead it sounds like maybe you need a tool that would take a VCF as input and produce summary statistics on the VCF that you could then decide on QC metrics for.  Does that sound like it would solve your problem?

**Comment by DarioS** [comment]
We run contamination checks on the somatic VCF using the AMBER / COBALT / PURPLE pipeline. It does not flag contamination. [PURPLE](https://github.com/hartwigmedical/hmftools/tree/master/purity-ploidy-estimator) is a purity and ploidy estimator but often struggles with low purity samples and erroneously reports a samples such as this as purity 1.00 and ploidy 2.00. I wonder if such a sample could be detected using some tool from GATK?

**Comment by fleharty** [comment]
@DarioS Sorry for the delay, have you tried CalculateContamination?

https://gatk.broadinstitute.org/hc/en-us/articles/360036888972-CalculateContamination

**Comment by DarioS** [comment]
Not yet. I am curious about the mathematical details of it. How does it distinguish no dominant clone from contamination or too many infiltrating immune cells? In the meantime, I wrote a small R script. Example output below. This is a sun-exposed skin cancer, hence the enormous mutation burden in whole genome sequencing data.

![image](https://user-images.githubusercontent.com/631218/87098600-08acb380-c28b-11ea-8876-9d9817ab13d7.png)

**Comment by davidbenjamin** [comment]
> How does it distinguish no dominant clone from contamination or too many infiltrating immune cells?

That's actually not hard because subclonal populations are not correlated with the population allele frequencies in the germline resource, while contaminating variants are.  CalculateContamination's statistical model accounts for this.  Infiltrating immune cells have the same variants as the patients germline and thus also don't affect it.

**Comment by davidbenjamin** [comment]
@DarioS Why don't you share your entire Mutect2 pipeline.

**Comment by DarioS** [comment]
This stage of the analysis was processed by the university's core facility. Looking at the folders, I see contamination checking was done by them and the values are very small (except for CSCC_0012-P1).

```
$ awk 'FNR == 2' *contamination.table
CSCC_0001-M1    0.001397910198920107    2.3931161011219502E-5
CSCC_0002-M1    0.0018110584523976473   3.0917637453135506E-5
CSCC_0003-M1    0.0021701013617243343   3.361091416043641E-5
CSCC_0004-M1    0.003547542544504977    4.217047661219818E-5
CSCC_0005-M1    0.002250251046380226    3.392538946825551E-5
CSCC_0006-M1    0.0012602899027448764   2.1547132944470522E-5
CSCC_0007-M1    0.00135679065051943     2.0692791516445317E-5
CSCC_0008-M1    0.004394182081805844    3.748225078434626E-5
CSCC_0009-M1    0.0019614948575730397   2.4555494660711082E-5
CSCC_0010-M1    0.004122282756273677    3.6210336627748355E-5
CSCC_0010-P1    0.001888852796306713    3.040210329291157E-5
CSCC_0011-M1    0.004616869166852859    3.9987828482322895E-5
CSCC_0012-M1    0.0013866025034395032   2.1835070542119526E-5
CSCC_0012-P1    0.9856060967650699      0.0023006992152694522
CSCC_0013-M1    0.014792148767770472    9.498068474793835E-5
CSCC_0014-M1    0.0028227703351458118   4.122117743000552E-5
CSCC_0015-M1    0.01467099675552882     9.531218938517413E-5
CSCC_0016-P1    0.0014411085088999514   2.716291906758587E-5
CSCC_0017-P1    0.0015213899870480127   2.712650453920576E-5
CSCC_0018-P1    0.001694677662867099    2.913615483470931E-5
CSCC_0019-P1    0.0016654868517623346   2.8602851235266697E-5
CSCC_0020-P1    0.0015496166402163914   2.7824601469663656E-5
```

The procedure done is

> Base quality score recalibration was performed with GATK 4.1.2.0 (Van der Auwera et al. 2013). GATK SplitIntervals was used to define 32 evenly-sized genomic intervals over which GATK BaseRecalibrator was run for each sample. The 32 recalibration tables per sample were merged into one table per sample with GATK GatherReports.
> 
> Base-recalibrated BAM files were produced with GATK ApplyBQSR in parallel over each of the 3,366 contigs in the hg 38 + alt reference genome, plus the unmapped reads. These were merged into a final BAM per sample with GATK GatherBamFiles.
> 
> SplitIntervals was used to define 3,200 evenly sized genomic intervals across hg38 + alt contigs, excluding telomeres, centromeres, unplaced contigs, unlocalized contigs, decoy and the Epstein-Barr viral sequence (Supplementary data 1). These intervals were used for scattering tasks in the Germline short variant calling and Somatic short variant calling workflows outlined below.
> 
> Germline short variant calling
> 
> HaplotypeCaller was run at the 3,200 pre-defined intervals for each sample and interval VCFs were gathered into GVCFs per sample. The following steps were then performed for the OSCC, TCGA and CSCC cohorts separately. Per sample GVCFs were consolidated with GenomicsDBImport and joint-called with GenotypeGVCFs at the 3,200 pre-defined intervals. Multi-sample interval calls were gathered using GatherVcfs. VariantFiltration and MakeSitesOnlyVcf was used to filter records with excess heterozygosity and retain only the site-level annotations. VariantRecalibrator and ApplyRecal were then applied for SNP and indels separately following GATK’s recommendations to obtain a final, filtered cohort VCF.
> 
> Somatic short variant calling
> 
> Somatic short variant calling was performed for tumour-normal pairs, first by preparing a panel of normals (PoN) for the OSCC, TCGA and CSCC cohorts separately. To create a PoN, Mutect2 was run in tumour only mode for the normal samples in the cohort at the 3,200 pre-defined intervals. GatherVcfs was used to gather interval VCFs into per sample VCFs. GenomicsDBImport and CreateSomaticPON was used to consolidate sample VCFs and create the PoN resource at the 3,200 pre-defined intervals. Interval VCFs were gathered into a single, cohort PoN VCF using GatherVcfs.
> 
>  
> 
> For each tumour-normal pair, Mutect2 was used for calling SNP and indels at the 3,200 pre-defined intervals, applying settings –f1r2-tar-gz, --germline-resource with gnomAD v3 as input and –panel-of-normals with PoN for the tumour-normal pair’s respective cohort as input. Interval stats files were merged into a single stats file with MergeMutectStats and interval VCFs were gathered with GatherVcfs into a single VCF for each tumour-normal pair, as preparation for further filtering using FilterMutectCalls.
> 
> We applied two additional workflows to prepare metrics and priors before running FilterMutectCalls. LearnReadOrientationModel was applied using f1r2 count metrics obtained from Mutect2 to calculate orientation bias priors. GetPileupSummaries and CalculateContamination tools were used to estimate cross sample contamination. Normal and tumour BAM files in the cohort and common biallelic SNP variants from the gnomAD v3 resource (obtained using SelectVariants with -select-type SNP -restrict-alleles-to BIALLELIC -select “AD > 0.05” –lenient settings applied) were used as inputs for GetPileupSummaries. We applied the -tumor-segmentation option in CalculateContamination.
> 
> A final, refined callset for each TN-pair was obtained by applying FilterMutectCalls to the tumor-normal VCFs with the –stats, --tumor-segmentation, --contamination-table and –ob-priors settings applied.
